pkg/agent/evictionmanager/plugin/memory/helper.go

```diff
@@ -0,0 +1,290 @@
+	action int, rankingMetrics []string, podToEvictMap map[string]*v1.Pod) {
+	filteredPods := e.filterPods(activePods, action)
+	if filteredPods != nil {
+		general.NewMultiSorter(e.getEvictionCmpFuncs(rankingMetrics, numaID)...).Sort(native.NewPodSourceImpList(filteredPods))
+		for i := 0; uint64(i) < general.MinUInt64(topN, uint64(len(filteredPods))); i++ {
+			podToEvictMap[string(filteredPods[i].UID)] = filteredPods[i]
+		}
+	}
+}
+
+func (e *EvictionHelper) filterPods(pods []*v1.Pod, action int) []*v1.Pod {
+	switch action {
+	case actionReclaimedEviction:
+		return native.FilterPods(pods, e.reclaimedPodFilter)
+	case actionEviction:
+		return pods
+	default:
+		return nil
+	}
+}
+
+// getEvictionCmpFuncs returns a comparison function list to judge the eviction order of different pods
+func (e *EvictionHelper) getEvictionCmpFuncs(rankingMetrics []string, numaID int) []general.CmpFunc {
+	cmpFuncs := make([]general.CmpFunc, 0, len(rankingMetrics))
+
+	for _, metric := range rankingMetrics {
+		currentMetric := metric
+		cmpFuncs = append(cmpFuncs, func(s1, s2 interface{}) int {
+			p1, p2 := s1.(*v1.Pod), s2.(*v1.Pod)
+			switch currentMetric {
+			case evictionconfig.FakeMetricQoSLevel:
+				isReclaimedPod1, err1 := e.reclaimedPodFilter(p1)
+				if err1 != nil {
+					klog.Errorf(errMsgCheckReclaimedPodFailed, p1.Namespace, p1.Name, err1)
+				}
+
+				isReclaimedPod2, err2 := e.reclaimedPodFilter(p2)
+				if err2 != nil {
+					klog.Errorf(errMsgCheckReclaimedPodFailed, p2.Namespace, p2.Name, err2)
+				}
+
+				if err1 != nil || err2 != nil {
+					// prioritize evicting the pod for which no error is returned
+					return general.CmpError(err1, err2)
+				}
+
+				// prioritize evicting the pod whose QoS level is reclaimed_cores
+				return general.CmpBool(isReclaimedPod1, isReclaimedPod2)
+			case evictionconfig.FakeMetricPriority:
+				// prioritize evicting the pod whose priority is lower
+				return general.ReverseCmpFunc(native.PodPriorityCmpFunc)(p1, p2)
+			default:
+				p1Metric, p1Found := e.getPodMetric(p1, currentMetric, numaID)
+				p2Metric, p2Found := e.getPodMetric(p2, currentMetric, numaID)
+				if !p1Found || !p2Found {
+					_ = e.emitter.StoreInt64(metricsNameFetchMetricError, 1, metrics.MetricTypeNameCount,
+						metrics.ConvertMapToTags(map[string]string{
+							metricsTagKeyNumaID: strconv.Itoa(numaID),
+						})...)
+					// prioritize evicting the pod for which no stats were found
+					return general.CmpBool(!p1Found, !p2Found)
+				}
+
+				// prioritize evicting the pod whose metric value is greater
+				return general.CmpFloat64(p1Metric, p2Metric)
+			}
+		})
+	}
+
+	return cmpFuncs
+
+}
+
+// getPodMetric returns the value of a pod-level metric.
+// And the value of a pod-level metric is calculated by summing the metric values for all containers in that pod.
+func (e *EvictionHelper) getPodMetric(pod *v1.Pod, metricName string, numaID int) (float64, bool) {
```

Comment:
we have a general function in metaserver, `AggregatePodNumaMetric`, can you use this to replace this self-defined logic here?

Reply:
the plugins  make different cmp functions according whether or not there is lost metrics. the `AggregatePodNumaMetric` won't return it for now. If we want reuse it here, we need to modify `AggregatePodNumaMetric`.